app/assets/javascripts/discourse/app/lib/intercept-click.js

```diff
@@ -3,12 +3,12 @@ import DiscourseURL from "discourse/lib/url";
 export function wantsNewWindow(e) {
   return (
     e.defaultPrevented ||
-    (e.isDefaultPrevented && e.isDefaultPrevented()) ||
+    e.isDefaultPrevented?.() ||
     e.shiftKey ||
     e.metaKey ||
     e.ctrlKey ||
-    (e.button && e.button !== 0) ||
-    (e.currentTarget && e.currentTarget.target === "_blank")
+    e.button ||
```

Comment:
```suggestion
    e.button !== 0 ||
```
I know it's functionally the same, but IMO this is way more readable. Otherwise it looks like we're checking for the presence of the button property, rather than its value.

Maybe even better would be to do

```js
const MOUSE_EVENT_PRIMARY_BUTTON_ID = 0;
```
and then

```js
e.button !== MOUSE_EVENT_PRIMARY_BUTTON_ID
```